main.cpp

```diff
@@ -73,12 +24,26 @@ void testPairToNumber(
     assert(pairNumber == expectedPairNumber);
 }
 
+
+void printPair(){
+   
+    for (int majorColor=0;majorColor<4;++majorColor){
+         for (int minorColor=0;minorColor<4;++minorColor){
```

Comment:
Avoid hard-coding numbers like 4. Take it from the length of the array.
BTW this loop goes through 4 minor and 4 major colors. There are actually five.